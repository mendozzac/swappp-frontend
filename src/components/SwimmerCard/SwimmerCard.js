import "./SwimmerCard.scss";

const SwimmerCard = ({ swimmer }) => {
  return (
    <li className="swimmer-card">
      <div className="swimmer-card_element">
        <img
          src={swimmer.image}
          alt={swimmer.name}
          width="140px"
          height="140px"
        />
        <span>{swimmer.name}</span>
        <span>{swimmer.surname}</span>
      </div>
    </li>
  );
};

export default SwimmerCard;
