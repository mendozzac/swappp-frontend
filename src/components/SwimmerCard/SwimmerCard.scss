@import "../../styles/extends";

.swimmer-card {
  width: 140px;
  height: 200px;
  font-size: 21px;
  font-weight: 600;
  @extend %center;
  cursor: pointer;
  margin: 20px;
  &_element {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
  }
}

img {
  border-radius: 50%;
}
