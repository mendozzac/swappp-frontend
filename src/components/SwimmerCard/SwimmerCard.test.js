import { render, screen } from "@testing-library/react";
import SwimmerCard from "./SwimmerCard";
import { BrowserRouter } from "react-router-dom";
import { MemoryRouter } from "react-router-dom";
import ReactTestRenderer from "react-test-renderer";

describe("Given a SwimmerCard component", () => {
  describe("When it receives swimmer information", () => {
    test("Then it should render a card with the information of the swimmer", () => {
      const swimmer = {
        name: "Geofrey",
        surname: "Campbell",
        birthdate: "11/08/90",
        height: 190,
        weight: 80,
        times: {
          distance: 200,
          style: "Espalda",
          date: "22/09/19",
          time: "1:56.09",
          pool: 50,
        },
      };

      render(
        <BrowserRouter>
          <SwimmerCard swimmer={swimmer} />
        </BrowserRouter>
      );
      const newSwimmer = screen.getByText("Geofrey");

      expect(newSwimmer).toBeInTheDocument();
    });
  });
  describe("When it receives a swimmer", () => {
    test("Then it should render a card with the swimmer", () => {
      const newSwimmer = {};

      const swimmerCard = ReactTestRenderer.create(
        <MemoryRouter>
          <SwimmerCard swimmer={newSwimmer} />
        </MemoryRouter>
      );

      expect(swimmerCard.toJSON()).toMatchSnapshot();
    });
  });
});
