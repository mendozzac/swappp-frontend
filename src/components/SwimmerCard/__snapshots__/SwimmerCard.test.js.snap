// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SwimmerCard component When it receives a swimmer Then it should render a card with the swimmer 1`] = `
<li
  className="swimmer-card"
>
  <div
    className="swimmer-card_element"
    onClick={[Function]}
  >
    <img
      alt="nadador"
      height="140px"
      width="140px"
    />
    <span />
    <span />
  </div>
</li>
`;
